src/node_file.cc

```diff
@@ -708,11 +684,7 @@ static void ReadLink(const FunctionCallbackInfo<Value>& args) {
                                                encoding,
                                                &error);
     if (rc.IsEmpty()) {
-      // TODO(addaleax): Use `error` itself here.
-      return env->ThrowUVException(UV_EINVAL,
-                                   "readlink",
-                                   "Invalid character encoding for link",
-                                   *path);
+      env->isolate()->ThrowException(error);
```

Comment:
`return env->isolate()->ThrowException(error);`?  The call to `rc.ToLocalChecked()` two lines down will fail if you fall through.

Same issue appears a few more times further down in this file.